pkgs/tools/filesystems/glusterfs/default.nix

```diff
@@ -36,18 +81,78 @@ rec {
 
   makeFlags = "DESTDIR=$(out)";
 
-  preInstall = ''
-    substituteInPlace api/examples/Makefile --replace '$(DESTDIR)' $out
-    substituteInPlace geo-replication/syncdaemon/Makefile --replace '$(DESTDIR)' $out
-    substituteInPlace geo-replication/syncdaemon/Makefile --replace '$(DESTDIR)' $out
-    substituteInPlace xlators/features/glupy/examples/Makefile --replace '$(DESTDIR)' $out
-    substituteInPlace xlators/features/glupy/src/Makefile --replace '$(DESTDIR)' $out
-    '';
+  enableParallelBuilding = true;
 
   postInstall = ''
     cp -r $out/$out/* $out
     rm -r $out/nix
-    wrapProgram $out/sbin/mount.glusterfs --set PATH "${stdenv.lib.makeBinPath [ coreutils gnused attr gnugrep which]}"
+    '';
+
+  postFixup = ''
+    # glusterd invokes `gluster` and other utilities when telling other glusterd nodes to run commands.
+    # For example for `peer_georep-sshkey` key generation, so `$out/bin` is needed in the PATH.
+    # It also invokes bash scripts like `gverify.sh`.
+    # It also invokes executable Python scripts in `$out/libexec/glusterfs`, which is why we set up PYTHONPATH accordingly.
+    # We set up the paths for the main entry point executables.
+
+    GLUSTER_PATH="${stdenv.lib.makeBinPath runtimePATHdeps}:$out/bin"
+    GLUSTER_PYTHONPATH="$(toPythonPath $out):$out/libexec/glusterfs"
+    GLUSTER_LD_LIBRARY_PATH="$out/lib"
+
+    wrapProgram $out/bin/glusterd --set PATH "$GLUSTER_PATH" --set PYTHONPATH "$GLUSTER_PYTHONPATH" --set LD_LIBRARY_PATH "$GLUSTER_LD_LIBRARY_PATH"
+    wrapProgram $out/bin/gluster --set PATH "$GLUSTER_PATH" --set PYTHONPATH "$GLUSTER_PYTHONPATH" --set LD_LIBRARY_PATH "$GLUSTER_LD_LIBRARY_PATH"
+    wrapProgram $out/sbin/mount.glusterfs --set PATH "$GLUSTER_PATH" --set PYTHONPATH "$GLUSTER_PYTHONPATH" --set LD_LIBRARY_PATH "$GLUSTER_LD_LIBRARY_PATH"
+
+    # Set Python environment for the Python based utilities.
+    # It would be nice if there was a better way to do this, automatically for all of them.
```

Comment:
Would checking for `#! …/python` be a good way to perform a sanity check that nothing obvious is missed?